Cannot use remote (production) server

Steps to repro:

1. Initialize named volume shodh-data on remote server
2. Initialize shodh-memory on remote server
3. Intitialize reverse proxy pointing to port 5910 
4. Use curl to retrieve health endpoint to verify connectivity
5. Update mcp settings for Claude Code
6. Launch Claude code with debug flag
7. check /mcp to verify Claude thinks it is connected
8. test the shodh-memory MCP server with: shodh-memory - memory_stats() (MCP)


----------

> ● I'll test the shodh-memory MCP server by calling the memory_stats function.
> 
> ● shodh-memory - memory_stats (MCP)
>   ⎿  Error: Memory server unavailable at http://127.0.0.1:3030. Please ensure shodh-memory-server is running.
> 
>      To start: cd shodh-memory && cargo run
> 
> ● The shodh-memory MCP server is not running. To test it, you need to start the server first:
> 
>   cd shodh-memory && cargo run
> 
>   Once the server is running on http://127.0.0.1:3030, the memory_stats() call will work. Would you like me to help you
>   start it, or would you prefer to run it yourself first?

----------


-----MCP Node-----
`
        "shodh-memory": {
            "command": "npx",
            "args": [
                "-y",
                "@shodh/memory-mcp"
            ],/
            "env": {
                "SHODH_PRODUCTION": "true",
                "SHODH_API_KEY": "<32 character API key>",
                "SHODH_API_URL": "https://<reverse-proxy address>",
                "SHODH_PORT": "443",
                "SHODH_HOST": "<reverse-proxy address>",
                "NODE_EXTRA_CA_CERTS": "<path to certchain>"
            }
        }
`

-----Initialization-----

> 2026-03-31T18:10:52.218Z [ERROR] MCP server "shodh-memory" Server stderr: [Stream] Initializing connection to ws://127.0.0.1:3030/api/stream
> [Stream] WebSocket error: ErrorEvent {
>   type: 'error',
>   defaultPrevented: false,
>   cancelable: false,
>   timeStamp: 80.479
> }
> [shodh-memory] Server binary not found. Please run: npx @shodh/memory-mcp
> [shodh-memory] Or download from: https://github.com/varun29ankuS/shodh-memory/releases
> Shodh-Memory MCP server v0.1.90 running
> Connecting to: http://127.0.0.1:3030
> User ID: claude-code
> Streaming: enabled
> Proactive surfacing: enabled
> 2026-03-31T18:10:52.218Z [DEBUG] MCP server "shodh-memory": Successfully connected to undefined server in 1443ms
> 2026-03-31T18:10:52.218Z [DEBUG] MCP server "shodh-memory": Connection established with capabilities: {"hasTools":true,"hasPrompts":true,"hasResources":true,"hasResourceSubscribe":false,"serverVersion":{"name":"shodh-memory","version":"0.1.90"}}
> 2026-03-31T18:10:52.218Z [DEBUG] [MCP] Server "shodh-memory" connected with subscribe=false
> 2026-03-31T18:10:55.244Z [DEBUG] MCP server "shodh-memory": Channel notifications skipped: server did not declare claude/channel capability


-----Calling shodh-memory-----

> 2026-03-31T18:36:20.604Z [DEBUG] MCP server "shodh-memory": Calling MCP tool: memory_stats
> 2026-03-31T18:36:20.607Z [DEBUG] [useDeferredValue] Messages deferred by 1 (5→6)
> 2026-03-31T18:36:20.615Z [DEBUG] [FileIndex] getProjectFiles called, respectGitignore=true
> 2026-03-31T18:36:20.615Z [DEBUG] [FileIndex] getFilesUsingGit called
> 2026-03-31T18:36:20.617Z [DEBUG] [useDeferredValue] Messages deferred by 1 (5→6)
> 2026-03-31T18:36:20.618Z [DEBUG] Fast mode unavailable: Fast mode requires extra usage billing · /extra-usage to enable
> 2026-03-31T18:36:20.622Z [DEBUG] AutoUpdaterWrapper: Installation type: native
> 2026-03-31T18:36:20.623Z [ERROR] MCP server "shodh-memory" Memory server unavailable at http://127.0.0.1:3030. Please ensure shodh-memory-server is running.
> 
> To start: cd shodh-memory && cargo run
> 2026-03-31T18:36:20.623Z [DEBUG] MCP server "shodh-memory": Tool 'memory_stats' failed after 0s: Memory server unavailable at http://127.0.0.1:3030. Please ensure shodh-memory-server is running.
> 
> To start: cd shodh-memory && cargo run
> 2026-03-31T18:36:20.625Z [DEBUG] mcp__shodh-memory__memory_stats tool error (22ms): Memory server unavailable at http://127.0.0.1:3030. Please ensure shodh-memory-server is running.
> 
> To start: cd shodh-memory && cargo run
> 2026-03-31T18:36:20.625Z [ERROR] McpToolCallError: McpToolCallError: Memory server unavailable at http://127.0.0.1:3030. Please ensure shodh-memory-server is running.
> 
> To start: cd shodh-memory && cargo run
>     at qyK (/$bunfs/root/src/entrypoints/cli.js:1826:104562)
>     at processTicksAndRejections (native:7:39)
> 2026-03-31T18:36:20.625Z [DEBUG] Broken symlink or missing file encountered for settings.json at path: /etc/claude-code/managed-settings.json
> 2026-03-31T18:36:20.629Z [DEBUG] LSP Diagnostics: getLSPDiagnosticAttachments called
> 2026-03-31T18:36:20.629Z [DEBUG] LSP Diagnostics: Checking registry - 0 pending



[shodh.yml](https://github.com/user-attachments/files/26386757/shodh.yml)

--- a/mcp-server/index.ts
+++ b/mcp-server/index.ts
@@ -36,7 +36,20 @@ const __filename = (typeof import.meta !== "undefined" && import.meta.url) ? fil
 const __dirname = __filename ? path.dirname(__filename) : process.cwd();
 
 // Configuration
-const API_URL = process.env.SHODH_API_URL || "http://127.0.0.1:3030";
+// Priority: SHODH_API_URL (full URL) > SHODH_HOST+SHODH_PORT (constructed) > localhost default
+function resolveApiUrl(): string {
+  if (process.env.SHODH_API_URL) return process.env.SHODH_API_URL;
+  const host = process.env.SHODH_HOST;
+  const port = process.env.SHODH_PORT;
+  if (host) {
+    const scheme = port === "443" ? "https" : "http";
+    const portSuffix = (port && port !== "443" && port !== "80") ? `:${port}` : "";
+    return `${scheme}://${host}${portSuffix}`;
+  }
+  if (port) return `http://127.0.0.1:${port}`;
+  return "http://127.0.0.1:3030";
+}
+const API_URL = resolveApiUrl();
 const WS_URL = API_URL.replace(/^http/, "ws") + "/api/stream";
 const USER_ID = process.env.SHODH_USER_ID || "claude-code";
 

